joomla/joomla-cms
# [4.3] Use CSRF only for the site's domain

Pull Request is alternative to #39343

### Summary of Changes

- CSRF is always sent with the `Joomla.request` ONLY for the site's domain
- The CSRF could be manually added for a foreign domain with a custom header `headers: {'X-CSRF-Token': Joomla.getOptions('csrf.token', '')} `

### Testing Instructions
- Go to `/administrator/index.php?option=com_config` and try to send a test mail, it should work (assuming your test site is properly set the mail server)
- The server should respond with a 200
- the url was modified to remove the form token, to make sure that CSRF was used
- Relative local domain modify [the url](https://github.com/joomla/joomla-cms/blob/226391c8e9375c763383cb7fb073871eaa85f8b4/administrator/components/com_config/tmpl/application/default_mail.php#L37) to:
```php
$ajaxUri = \Joomla\CMS\Uri\Uri::base(true) . '/index.php?option=com_config&task=application.sendtestmail&format=json';
```
refresh the page and run click on the send test mail again, observe the request (in the browser console) that has the x-csrf

![Screenshot 2023-01-09 at 17 12 57](https://user-images.githubusercontent.com/3889375/211357120-f550fbbb-7196-4469-827f-aebfd27a9845.png)

- Absolute local domain modify [the url](https://github.com/joomla/joomla-cms/blob/226391c8e9375c763383cb7fb073871eaa85f8b4/administrator/components/com_config/tmpl/application/default_mail.php#L37) to:
```php
$ajaxUri = \Joomla\CMS\Uri\Uri::base(false) . '/index.php?option=com_config&task=application.sendtestmail&format=json';
```
refresh the page and run click on the send test mail again, observe the request (in the browser console) that has the x-csrf


- Foreign domain modify [the url](https://github.com/joomla/joomla-cms/blob/226391c8e9375c763383cb7fb073871eaa85f8b4/administrator/components/com_config/tmpl/application/default_mail.php#L37) to:
```php
$ajaxUri = 'https://www.joomla.org/4/administrator/index.php?option=com_config&task=application.sendtestmail&format=json';
```
refresh the page and run click on the send test mail again, observe the request (in the browser console) that has the x-csrf
![Screenshot 2023-01-09 at 17 24 04](https://user-images.githubusercontent.com/3889375/211356982-9d61390d-2bd5-4d0d-a4d4-c251efb9c4c4.png)

### Actual result BEFORE applying this Pull Request

CSRF send to all domains

### Expected result AFTER applying this Pull Request

CSRF is only for local use

### Link to documentations
Please select:
